docs/CloudSpanner.md

```diff
@@ -0,0 +1,64 @@
+Trillian on Cloud Kubernetes with Cloud Spanner
+====================================================
+
+This document outlines the steps necessary to get an instance of Trillian up
+and running on Google Cloud.
+
+ 1. Create a Google Cloud project (e.g. trillian-opensource-ci)
+ 2. Enable APIs
+    1. Kubernetes
+    2. Cloud Spanner
+    3. ...
+
+ 3. Create the public API key (see https://cloud.google.com/kubernetes-engine/docs/tutorials/authenticating-to-cloud-platform)
+   1. From the APIs & Services > Credentials page, click Create credentials > API key.
+      A key will automatically be created.
+   2. Restrict API key to only have access to:
+      * Google Cloud APIs
+      * Cloud Spanner API
+   3. Download service account key (we'll call that file `service-key.json`)
+   4. run: `kubectl create secret generic spanner-key --from-file=key.json=service-key.json`
+
+ 4. Create Spanner instance & database
+   1. Click on menu > Cloud Spanner
+   2. Click on "Create Instance"
+   3. Use instance id "trillian-opensource-ci"
+   4. Choose region (e.g. us-central1)
+   5. Choose number of nodes
+   6. Click create Database
+     1. Use DB name "trillian-opensource-ci-db"
+     2. Click continue
+     4. In `Define your database schema`, click the `Edit as text` slider
+     3. paste contents of [spanner.sd](storage/cloudspanner/spanner.sdl) into the text box (you may need to remove the SQL comments prefixed with `--` at the top)
+     4. Click on create
+
+ 5. Create kubernetes cluster
+   1. menu > Kubernetes
+   2. click on Create Cluster
+   3. Set cluster name to something (we'll call this `${CLUSTER_NAME}`)
+   4. Set zone to us-central1-a
```

Comment:
This assumes that's what they picked above.

Reply:
changed to a ${var}